Remove temporary workspaces created by Abins when calling SaveAscii

ajjackson opened the pull request:
Abins uses the algorithm function "Scale" to re-scale the data on-the-fly when calling SaveAscii. By default this function call creates a hidden workspace which is not cleaned up.

**Purpose of work**
As identified in beta testing, when using Abins with hidden workspaces enabled, an unnecessary workspace are created as a side effect each time Abins is used with SaveAscii=True

**Summary of work**
Here we:

- Set StoreInADS=False when calling Scale, so the temporary workspace doesn't appear to the user, even if they have hidden workspaces visible
- Delete the workspace afterwards to free memory

**To test:**

On the main branch, use the Settings dialogue to "Show Invisible Workspaces". Run Algorithms/Simulation/Abins:

- set VibrationalOrPhononFile to one of the .phonon files from UnitTest data
- set AbInitioProgram to "CASTEP"
- set an output workspace
- make sure "SaveAscii" is enabled
- Other settings shouldn't matter, defaults are fine.

A mysterious workspace with a leading underscore should appear.

Repeat the process on this branch; there should be no such workspace.

Fixes #35578 

A small release note has been added.

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

ajjackson committed:
Release note

robertapplin commented on scripts/abins/abinsalgorithm.py:
```diff
@@ -613,7 +613,13 @@ def write_workspaces_to_ascii(scale: float = 1.0, *, ws_name: str) -> None:
         for wrk_num in range(num_workspaces):
             wrk = mtd[ws_name].getItem(wrk_num)
-            SaveAscii(InputWorkspace=Scale(wrk, scale, "Multiply"), Filename=wrk.name() + ".dat", Separator="Space", WriteSpectrumID=False)
+            SaveAscii(
+                InputWorkspace=(tmp_workspace := Scale(wrk, scale, "Multiply", StoreInADS=False)),
+                Filename=wrk.name() + ".dat",
+                Separator="Space",
+                WriteSpectrumID=False,
+            )
+            DeleteWorkspace(tmp_workspace)
```
The `DeleteWorkspace` algorithm only removes the workspace from the ADS if it exists in the ADS. Seeing as StoreInADS=False, the workspace does not exist in the ADS - and so we shouldn't need to use `DeleteWorkspace`

This should be enough for it to work:
```suggestion
            SaveAscii(
                InputWorkspace=Scale(wrk, scale, "Multiply", StoreInADS=False),
                Filename=wrk.name() + ".dat",
                Separator="Space",
                WriteSpectrumID=False,
            )
```

robertapplin replied:
Nice use of the walrus operator - I didn't realise it was in python 3.8!

ajjackson replied:
We should do some kind of announcement each time the minimum version requirement changes: "here are the neat Python features you can, and probably should, start using" :smile:

AnthonyLim23 replied:
@ajjackson is that you volunteering to give a code chat on some nice new toys (features)?

ajjackson replied:
I suspect others are well-qualified for this but maybe :wink:

ajjackson committed:
Remove unecessary DeleteWorkspace

If it's not in the ADS and not assigned to a variable, it shouldn't exist, right?

*looks suspiciously at garbage collector*

Co-authored-by: Rob Applin <40830825+robertapplin@users.noreply.github.com>

robertapplin commented:
Assuming this is going into `release-next`, we need to change the base branch:
https://developer.mantidproject.org/GitWorkflow.html#fixing-a-pr-with-an-incorrect-base-branch

robertapplin reviewed (approved):
Thanks for the changes, looks good to me 👍
